### DISCUSSION TITLE
Help install & use

### DISCUSSION THREAD

**Discussion by moreje** [iantrich/restriction-card#109] [General]
Hello, I don't know if it is an issue or mise use... so I ask here:
I installed the module through HACS:
![image](https://github.com/iantrich/restriction-card/assets/5374164/7a1ef809-1db6-4233-91f5-b48fe2e66caa)
then went to my dashboard, I checked if the ressources list restriction-card module:
![image](https://github.com/iantrich/restriction-card/assets/5374164/310c8573-f572-4681-9a16-c0fd0bfb0ea2)
but when adding a new card, can't find anything about restriction card:
![image](https://github.com/iantrich/restriction-card/assets/5374164/9648efd7-8356-4599-ba9e-3490d59861e5)
so what am I doing wrong??
thank you for your help

**Comment by ildar170975** [comment]
As many custom cards, the restriction-card does not have own UI editor.
Add ANY card, switch to yaml - and then type the code for the restriction-card.

Alternatively:
Assume you need to have Entities card inside restriction-card.
Then add the Entities card from UI, then switch to yaml - and add a code for the restriction-card.

**Reply by moreje** [reply]
ok, thank you. I understand I must edit the YAML code of my card, but don't know how:
I want to protect a button card, so I add this one:
![image](https://github.com/iantrich/restriction-card/assets/5374164/b0ba9b34-e7e8-4b31-a132-b676e1158dbb)
then switch to yaml:
![image](https://github.com/iantrich/restriction-card/assets/5374164/b268c913-11da-4248-a360-d01a47c0940a)

what am I supposed to write in the code? and where?

**Comment by ildar170975** [comment]
There are examples here, have you tried to reproduce them?
https://github.com/iantrich/restriction-card?tab=readme-ov-file#example-configurations

**Reply by moreje** [reply]
OK, I found my way!
I did not understood I had to add card using manual option, and then replace the entity...
I thought I had first to add a card...and then edit its yaml code....
now I can use the examples and test!
thank you

**Reply by ildar170975** [reply]
> I thought I had first to add a card...and then edit its yaml code....

You surely can do it.
Assume you need to have Entities card with 10 rows inside restriction-card.
Then add the Entities card via UI, customize it as you wish.
Now you got a 100% ready Entities card.
Then go to yaml - and add all required restriction-card code:
from
```
type: entities
entities:
 - ....
```
to
```
... your restriction-card code ...
card:
  ... your already prepared inner card:
  type: entities
  entities:
   - ....
```

**Reply by Jaguaza** [reply]
> > I thought I had first to add a card...and then edit its yaml code....
> 
> You surely can do it. Assume you need to have Entities card with 10 rows inside restriction-card. Then add the Entities card via UI, customize it as you wish. Now you got a 100% ready Entities card. Then go to yaml - and add all required restriction-card code: from
> 
> ```
> type: entities
> entities:
>  - ....
> ```
> 
> to
> 
> ```
> ... your restriction-card code ...
> card:
>   ... your already prepared inner card:
>   type: entities
>   entities:
>    - ....
> ```

I'm sorry to bother you but I'm not clear.

I have this card:

```
type: custom:mushroom-template-card
entity: input_boolean.on_off
primary: ON/OFF
icon: |-
  {% if is_state('sensor.status', 'online') %}
    mdi:motion-play-outline
  {% elif is_state('sensor.status', 'connected') %} 
    mdi:motion-play-outline
  {% else %}
    mdi:motion-pause-outline
  {% endif %}
icon_color: |-
  {% if is_state('sensor.status', 'online') %}
  blue
  {% elif is_state('sensor.status', 'connected') %} 
  red
  {% endif %}
secondary: |-
  {% if is_state('sensor.status', 'online') %}
  Conectado
  {% elif is_state('sensor.status', 'connected') %}
  Cargando
  {% else %}
  Apagado
  {% endif %}
layout: horizontal
fill_container: true
multiline_secondary: false
tap_action:
  action: toggle
hold_action:
  action: more-info
```


How can I add a Pin. Thank you for everything.

**Reply by ildar170975** [reply]
Try smth like this:
```
type: custom:restriction-card
restrictions:
  pin:
    code: 1234
card:
  type: custom:mushroom-template-card
  entity: input_boolean.on_off
  ... your code for mushroom ...
```

**Reply by Jaguaza** [reply]
> Try smth like this:
> 
> ```
> type: custom:restriction-card
> restrictions:
>   pin:
>     code: 1234
> card:
>   type: custom:mushroom-template-card
>   entity: input_boolean.on_off
>   ... your code for mushroom ...
> ```

Thanks!!!!!

**Comment by brucecleghorn** [comment]
Hi Guys

Great Card, I've used it a lot to ask for Pin entry and plain Ack on many of my front end cards. I love it, thanks so much

However, I'm failing over and over again where I just want to add a restriction to a single entity within a card, and all the other entities behave as if the restriction card was not there. The Use Case is where I "NEVER" want to accidently turn off an external Pond Pump, but all the other entities can be toggled on/off

Could you give me any pointers please
Thanks so much
Bruce

type: entities
entities:
  - entity: light.living_room_floor_lamp
  - entity: light.living_room_table_lamp
  - entity: switch.living_room_wall_lights_switch_1
  - entity: switch.zigbee_ikea_switch_pond_filter_pump_switch
title: Living Room
state_color: true

**Reply by ildar170975** [reply]
1. Please always post a code properly formatted (placed in triple "`" backticks).
2. Strongly suggest you to use the main Community thread for asking questions: https://community.home-assistant.io/t/lovelace-restriction-card-client-side-security/142889.

As a starting point: if you wish to protect a particular entity in Entities card - use this:
```
type: entities
entities:
  - entity: switch.xxx
  - entity: switch.yyy
  - type: custom:restriction-card
    row: true
    card:
      entity: switch.zzz
```

**Reply by ildar170975** [reply]
Check examples in Docs.

**Reply by brucecleghorn** [reply]
Thanks so much mate
BTW, If your username is your birthday, we share the same birthday. HAHAHAHA. Although I'm 8 years old than you
Thanks mate, that worked

**Reply by ildar170975** [reply]
Good that you sorted this out!
